microprot/scripts/process_fasta.py

```diff
@@ -123,11 +126,57 @@ def read_representatives(represent):
     return sorted(list(ids))
 
 
+def split_fasta(seqs, prefix=None, outdir=None):
+    """Split a multi-protein FASTA file into single-sequence FASTAs
+
+    Parameters
+    ----------
+    seqs : list of skbio.sequence
+        List of skbio protein sequences with a protein name in header
+    prefix : str
+        Name prefix to be added to output single-sequence FASTA files
+
+    Raises
+    ------
+    TypeError
+        seqs needs to be a filepath or skbio.sequence object
+    """
+
+    if isinstance(seqs, str):
+        if os.path.exists(seqs):
+            seqs = extract_sequences(seqs)
+        else:
+            raise TypeError('split_fasta sequence input is not a filepath or '
+                            'filepath does not exist.')
+    elif isinstance(seqs, list):
+        if len(seqs) == 0:
+            raise ValueError('Empty list provided to split_fasta')
+        else:
+            if not isinstance(seqs[0], Sequence):
+                raise TypeError('Object you provided to split_fasta is not '
+                                'a skbio.sequence object')
+    else:
+        raise TypeError('split_fasta input sequences need to be a filepath or'
+                        'skbio.sequence object.')
+
+    if not outdir:
```

Comment:
I didn't want to make `outfile` required, because by default if you run
```
python process_fasta.py -i file.fasta --split
```
you get your output in cwd. I can change outfile to required, so that it would nee to be
```
python process_fasta.py -i file.fasta --split -o ./
```
at least.